app/Platform/Data/TicketData.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Data;
+
+use App\Models\Ticket;
+use App\Platform\Enums\TicketableType;
+use Spatie\LaravelData\Data;
+use Spatie\LaravelData\Lazy;
+use Spatie\TypeScriptTransformer\Attributes\TypeScript;
+
+#[TypeScript('Ticket')]
+class TicketData extends Data
+{
+    public function __construct(
+        public int $id,
+        public TicketableType $ticketableType,
+        public Lazy|int $state,
+        public Lazy|AchievementData|LeaderboardData|GameData $ticketable,
```

Comment:
Small prep here in this new `Ticket` DTO for multiple ticketable types. This is just future-thinking ... everything in `fromTicket()` hardcodes to `TicketableType` being an achievement.